tests/test_java.py

```diff
@@ -718,3 +720,284 @@ def test_java_frames_include_no_semicolons(
+        ),
+        (
+            "hotspot-jdk-11",
+            [
+                {"name": "CICompilerCount", "type": "intx", "value": "4", "origin": "ergonomic", "kind": ["product"]},
+                {"name": "ConcGCThreads", "type": "uint", "value": "2", "origin": "ergonomic", "kind": ["product"]},
+                {
+                    "name": "G1ConcRefinementThreads",
+                    "type": "uint",
+                    "value": "8",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "G1HeapRegionSize",
+                    "type": "size_t",
+                    "value": "1048576",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "GCDrainStackTargetSize",
+                    "type": "uintx",
+                    "value": "64",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "InitialHeapSize",
+                    "type": "size_t",
+                    "value": "264241152",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MarkStackSize",
+                    "type": "size_t",
+                    "value": "4194304",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MaxHeapSize",
+                    "type": "size_t",
+                    "value": "4196401152",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MaxNewSize",
+                    "type": "size_t",
+                    "value": "2517630976",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MinHeapDeltaBytes",
+                    "type": "size_t",
+                    "value": "1048576",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "SegmentedCodeCache",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "UseCompressedClassPointers",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "ergonomic",
+                    "kind": ["lp64_product"],
+                },
+                {
+                    "name": "UseCompressedOops",
+                    "type": "bool",
+                    "value": "true",
```

Comment:
Why do you specify the `value` everywhere if you gonna `pop` it before comparing?

Instead - and because we should probably test the collection of `value` - how about you don't store the `value` for flags which might change due to ergonomics, and when `value` is not present, `assert_jvm_flags_equal` will not test it, but when present it will test?

Reply:
we agreed that its not that important anyway, I would prefer leave it as is, as it checks default flag collection